### ISSUE TITLE
[Enhancement] Native Razor/Blazor format as an alternative for MVVM and xaml 

### ISSUE THREAD

**Issue by meirkr** [xamarin/Xamarin.Forms#7552]
## Summary
Let developers build the xamarin native forms and controls using Razor methodology.

## Intended Use Case
Instead of creating MVVM classes and xaml for defining controls and binding, support creating razor pages and components with the same binding and components nesting capabilities.

This proposal is different than the existing github project "BlazorMobile" (https://github.com/Daddoon/BlazorMobile) which uses WebView to host a web assembly blazor pages inside Xamarin app.

Instead, the xamarin will build native controls and native runtime similar to the way native script (js/angular/Vue) does.

**Comment by Eilon** [comment]
Hi @meirkr , we are looking at various options right now to integrate these technologies to provide more options for people. We don't have anything we can share right now, but we'd love to hear any thoughts you have on this to help guide us towards possible product features.

**Comment by meirkr** [comment]
Hi

I don't understand whether I have missed some details.
I just wanted to share a new proposal for this idea, as anyone talks about SPA which is being transformed into cross native mobile applications development with same look and feel, but still mobile native: https://www.nativescript.org.

So, as .NET-Foundation/Microsoft pushes the new SPA - **Blazor**, why not having the same here?
I have an impression that XAML and MVVM are going to be obsolete, as moving to web based SPA development, including being hosted by "Electrone". for desktop apps This is my feeling, after all...

So over time, I think people will like less to learn XAML and MVVM, as they would like to write SPA based apps (In my work we have moved from WPF to Angular as well).
For example - I have started to learn angular, and I see angular works also as native platform under typescript as linked above in this comment.
Basically, I love .NET, more than angular (JS based).
And Blazor tries to present this new competitor to existing web based SPA.
Also, there is the BlazorMobile project which tries to bring that SPA web based solution tweeked into a Xamarin app.
But hinking about it, why doing such tweek? Why not bringing Blazor/Razor pages and components to the native world of Xamarin? without web assembly and web view, just natively, as it works using the XAML/MVVM?
Wouldn't it be better as attractive new technology which will reuse pages/code in all platform natively - whether it is web or android/OS?

What do you think about this idea?

**Comment by Eilon** [comment]
@meirkr I don't think you've missed any details. NativeScript is certainly interesting, and within the team here we have some similar ideas. The Blazor model is an MVU model (Model-View-Update), so it's similar to other popular SPA frameworks. Blazor also uses an HTML-like syntax, which some people prefer.

Here's an example of the type of syntax we're looking at:

```html
<StackLayout Orientation="StackOrientation.Horizontal">
    <Button Text="+1" OnClick="@HandleClick" />
    <Label FontSize="30" FontAttributes="FontAttributes.Italic" Text="@("You have pressed " + AppState.Counter + " times")" />
</StackLayout>

@code {
    void HandleClick()
    {
        AppState.Counter++;
    }
}
```

It's using Blazor syntax, but the elements are what you expect from Xamarin.Forms, which already has a rich set of cross-platform native components.

There's another project called Comet that you can find here: https://github.com/Clancey/Comet/ - it also provides an MVU pattern for building cross-platform apps.

**Comment by samhouts** [comment]
Please see https://github.com/xamarin/MobileBlazorBindings. Thanks!
